samples/bluetooth/central_hr_coded/src/main.c

```diff
@@ -0,0 +1,277 @@
+{
+	bt_addr_le_t *addr = user_data;
+	int i;
+	struct bt_conn_le_create_param *conn_params;
+
+	printk("[AD]: %u data_len %u\n", data->type, data->data_len);
+
+	switch (data->type) {
+	case BT_DATA_UUID16_SOME:
+	case BT_DATA_UUID16_ALL:
+		if (data->data_len % sizeof(uint16_t) != 0U) {
+			printk("AD malformed\n");
+			return true;
+		}
+
+		for (i = 0; i < data->data_len; i += sizeof(uint16_t)) {
+			struct bt_uuid *uuid;
+			uint16_t u16;
+			int err;
+
+			memcpy(&u16, &data->data[i], sizeof(u16));
+			uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(u16));
+			if (bt_uuid_cmp(uuid, BT_UUID_HRS)) {
+				continue;
+			}
+
+			err = bt_le_scan_stop();
+			if (err) {
+				printk("Stop LE scan failed (err %d)\n", err);
+				continue;
+			}
+
+			conn_params = BT_CONN_LE_CREATE_PARAM(
+					BT_CONN_LE_OPT_CODED | BT_CONN_LE_OPT_NO_1M,
+					BT_GAP_SCAN_FAST_INTERVAL,
+					BT_GAP_SCAN_FAST_INTERVAL);
+
+			err = bt_conn_le_create(addr, conn_params,
+					BT_LE_CONN_PARAM_DEFAULT,
+					&default_conn);
+			if (err) {
+				printk("Create conn failed (err %d)\n", err);
+				start_scan();
+			}
+
+			printk("Connection pending\n");
+
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static void device_found(const struct bt_le_scan_recv_info *info,
+		struct net_buf_simple *ad)
+{
+	char dev[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(info->addr, dev, sizeof(dev));
+	printk("[DEVICE]: %s, AD type %u, AD props %u, AD data len %u, RSSI %i, ",
+			dev, info->adv_type, info->adv_props, ad->len, info->rssi);
+	printk("Primary PHY %u, Secondary PHY %u\n",
+			info->primary_phy, info->secondary_phy);
+
+	bt_data_parse(ad, eir_found, (void *)info->addr);
+}
+
+static struct bt_le_scan_cb scan_cb = {
+	.recv = device_found
+};
+
+static void start_scan(void)
+{
+	int err;
+
+	/* Use active scanning and disable duplicate filtering to handle any
+	 * devices that might update their advertising data at runtime. */
+	struct bt_le_scan_param scan_param = {
+		.type     = BT_LE_SCAN_TYPE_ACTIVE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window   = BT_GAP_SCAN_FAST_WINDOW,
+		.options  = BT_LE_SCAN_OPT_CODED | BT_LE_SCAN_OPT_NO_1M
+	};
+
+	err = bt_le_scan_start(&scan_param, NULL);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	int err;
+	struct bt_conn_info info;
+	const struct bt_conn_le_phy_info *phy_info;
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("Failed to connect to %s (%u)\n", addr, conn_err);
+
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+
+		start_scan();
+		return;
+	}
+
+	err = bt_conn_get_info(conn, &info);
+	if (err) {
+		printk("Failed to get connection info\n");
+	}
+	phy_info = info.le.phy;
+
+	printk("Connected: %s, tx_phy %u, rx_phy %u\n",
+			addr, phy_info->tx_phy, phy_info->rx_phy);
```

Comment:
If get_info returns err then phy_info->tx_phy/rx_phy will not be valid.
moved this into an else of the above statement. Can move const struct bt_conn_le_phy_info *phy_info; declaration down as well then.